### PR TITLE
Create signup&login endpoint, token verification and middleware

### PR DESCRIPTION
**Pull request by QianyingLuo**

### CODE REVIEW

File: src/assets/pages/index.html [hunk 1]
```diff
@@ -12,6 +12,14 @@
     </div>
 </div>    
 
+<script>
+    document.addEventListener("DOMContentLoaded", function() {
+        const token = document.cookie.split('; ').find(row => row.startsWith('token='));
+            if (token) {
+                console.log("Tengooo token")
```
Comment (devs-toni) [inline src/assets/pages/index.html]: mmm... and this?¿?

File: src/assets/pages/index.html [hunk 2]
```diff
@@ -34,7 +42,7 @@ <h5>{{ product.name | truncate(120) }}</h5>
             <p>
                 {% if product.discount_percentage != 0 %}
                     <span style="text-decoration: line-through;">{{ product.price }}€</span><br>
-                    <span style="color: rgb(227, 43, 43)">{{ product.discounted_price }}€</span> 
+                    <span style="color: rgb(227, 43, 43)">{{ (product.price * (1 - product.discount_percentage)) | round(2) }}€</span>
```
Comment (devs-toni) [inline src/assets/pages/index.html]: Didn't we already calculate this in the backend?

File: src/controller/index.py [hunk 3]
```diff
@@ -11,8 +12,14 @@
 templates.env = ExtendedEnvironment(loader=FileSystemLoader("src/assets/"))
 templates.env.filters['truncate'] = templates.env.truncate
 
+SECRET_KEY = os.environ.get("SECRET_KEY")
+ALGORITHM = os.environ.get("ALGORITHM")
```
Comment (devs-toni) [inline src/controller/index.py]: If this happens again (which it does ...) you have to create a file in the path /src/config/config.py with the reading of these variables

File: src/controller/login.py [hunk 4]
```diff
@@ -1,13 +1,47 @@
...
+from ..api import user as user_api
+from ..domain import user as user_domain
+from ..useful import verify_token
 
 router = APIRouter()
+templates = Jinja2Templates(directory="src/assets/")
+
+SECRET_KEY = os.environ.get("SECRET_KEY")
+ALGORITHM = os.environ.get("ALGORITHM")
+ACCESS_TOKEN_EXPIRE_MINUTES = int(os.environ.get("ACCESS_TOKEN_EXPIRE_MINUTES", 1))
```
Comment (devs-toni) [inline src/controller/login.py]: d

File: src/controller/offers.py [hunk 5]
```diff
@@ -20,3 +20,38 @@ def __init__(self, **kwargs):
...
+        "Monitores": "/category/monitors"
+    }
+
+    return templates.TemplateResponse("pages/index.html", {
+        "request": request,
+        "products_by_type": products_by_type,
+        "category_urls": category_urls
+    })
+
+def check_token(request: Request) -> bool:
```
Comment (devs-toni) [inline src/controller/offers.py:45]: this function will you never use it again elsewhere?

**Review by devs-toni — APPROVED** [review]

File: src/assets/pages/index.html (hunk 1 above)
Comment (QianyingLuo) [reply src/assets/pages/index.html]: Done

File: src/assets/pages/index.html (hunk 2 above)
Comment (QianyingLuo) [reply src/assets/pages/index.html]: Done

File: src/controller/index.py (hunk 3 above)
Comment (QianyingLuo) [reply src/controller/index.py]: Done

File: src/controller/login.py (hunk 4 above)
Comment (QianyingLuo) [reply src/controller/login.py]: Done

File: src/controller/offers.py (hunk 5 above)
Comment (QianyingLuo) [reply src/controller/offers.py:45]: Done

**Pull request merged**